src/coreclr/vm/exceptionhandling.cpp

```diff
@@ -3185,7 +3185,7 @@ extern "C" void * QCALLTYPE CallCatchFunclet(QCall::ObjectHandleOnStack exceptio
 #ifdef TARGET_AMD64
 #ifdef TARGET_UNIX
             pvRegDisplay->pCurrentContext->Rdi = dwResumePC;
-            pvRegDisplay->pCurrentContext->Rsi = GetIP(pThread->GetAbortContext()); //How can this ever be different from dwResumePC?
+            _ASSERTE(GetIP(pThread->GetAbortContext()) == dwResumePC);
```

Comment:
Should this assert be enabled for all platforms (ie before `#ifdef TARGET_AMD64` )?

Reply:
The Unix amd64 was ever the only platform that was passing in the 2nd argument (even with the old EH). I cannot see how it could have ever been different than the dwResumePC and adding the 2nd one was actually my fault. I have added support for thread abort at catch resume for Unix two years ago and this was part of it. So this assert is just to make me feel safer. I would be fine not having it at all.

Reply:
I think we should either not have the assert, or move it up to be arch neutral, or have a comment why this invariant holds on amd64 only.